bin.js

```diff
@@ -6,25 +6,28 @@ const maxShowTime = 1000;
 let levelH = buffer * 1.7;
 
 class Bin {
-  constructor(w, i, goal) {
+  constructor(w, i, goal, levels) {
     this.w = w;
     this.i = i;
     this.x = i * w + w * 0.5;
     this.y = height - buffer * 0.75;
-    this.count = 0;
     this.goal = goal;
 
     this.levelGoal = this.goal / 4;
 
     this.levelsYOffset = levelH;
 
-    this.levels = {
+// if levels is undefined, assign empty levels
+    this.levels = levels ?? {
```

Comment:
Use the nullish coalescing operator to assign empty levels if no levels are passed in